pkg/install/cleanup.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2021 k0s Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package install
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/k0sproject/k0s/internal/util"
+	"github.com/sirupsen/logrus"
+)
+
+type CleanUpConfig struct {
+	containerdSockerPath string
+	criSocketPath        string
+	crictlBinPath        string
+	dataDir              string
+	containerdBinPath    string
+	quit                 chan bool
+	runDir               string
+}
+
+func (c *CleanUpConfig) WorkerCleanup() error {
+	var msg []string
+
+	if err := c.workerPreFlightChecks(); err != nil {
+		logrus.Fatalf("failed clean up pre-flight-checks: %v", err)
+	}
+	logrus.Info("starting containerd for cleanup operations...")
+
+	quit := make(chan bool)
+	c.quit = quit
```

Comment:
```suggestion
	c.quit := make(chan bool)
```

Reply:
> Suggested change 
> 	quit := make(chan bool)
> 	c.quit = quit
> 	c.quit := make(chan bool)

This doesn't actually work, because then the compiler complains that `c.quit evaluated but not used`